app/src/app/components/Toolbar/SaveMapModal.tsx

```diff
@@ -0,0 +1,327 @@
+import {useMapStore} from '@/app/store/mapStore';
+import React, {useMemo, useEffect} from 'react';
+import {
+  Button,
+  Flex,
+  Text,
+  Box,
+  TextField,
+  RadioGroup,
+  TextArea,
+  Blockquote,
+  Dialog,
+} from '@radix-ui/themes';
+import {Cross2Icon} from '@radix-ui/react-icons';
+import type {DocumentMetadata} from '../../utils/api/apiHandlers/types';
+import {styled} from '@stitches/react';
+import {checkoutDocument} from '@/app/utils/api/mutations';
+import {saveMap} from '@/app/utils/api/apiHandlers/saveMap';
+import {useMapStatus} from '@/app/hooks/useMapStatus';
+import {useMapMetadata} from '@/app/hooks/useMapMetadata';
+
+const BoxContainer = styled(Box, {
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '1rem',
+});
+
+const DialogContentContainer = styled(Dialog.Content, {
+  maxHeight: 'calc(100vh-2rem)',
+  maxWidth: '60vw',
+  overflowY: 'auto',
+});
+
+export const SaveMapModal: React.FC<{
```

Comment:
Low: At some point in the future we may want to refactor some of this state to a custom hook or zustand store